internal/report/source.go

```diff
@@ -214,14 +215,73 @@ func PrintWebList(w io.Writer, rpt *Report, obj plugin.ObjTool, maxFiles int) er
 		}
 
 		printFunctionHeader(w, ff.functionName, path, n.Flat, n.Cum, rpt)
+		percentiles := calculatePercentiles(fnodes)
 		for _, fn := range fnodes {
-			printFunctionSourceLine(w, fn, asm[fn.Info.Lineno], reader, rpt)
+			printFunctionSourceLine(w, fn, asm[fn.Info.Lineno], reader, getPercentileString(float64(fn.Cum), percentiles), rpt)
 		}
 		printFunctionClosing(w)
 	}
 	return nil
 }
 
+func getPercentileString(cumSum float64, percentiles map[float64]float64) string {
+	if cumSum == 0 {
+		return ""
+	}
+	switch {
+	case cumSum >= percentiles[80]:
```

Comment:
I have implemented these changes. The code now calculates the percentile on the array of cum values, instead of a set. I'm not using float64s anymore, instead I use int64. This last change meant that I had to implement the Sort Interface for int64.